dbcon/joblist/tuple-bps.cpp

```diff
@@ -2427,93 +2439,96 @@ void TupleBPS::receiveMultiPrimitiveMessages(uint32_t threadID)
 
         } // done reading
 
-    }//try
+    }  // try
     catch (...)
     {
-        handleException(std::current_exception(),
-                        logging::ERR_TUPLE_BPS,
-                        logging::ERR_ALWAYS_CRITICAL,
-                        "st: " + std::to_string(fStepId) +
-                            " TupleBPS::receiveMultiPrimitiveMessages()");
+        handleException(std::current_exception(), logging::ERR_TUPLE_BPS, logging::ERR_ALWAYS_CRITICAL,
+                        "st: " + std::to_string(fStepId) + " TupleBPS::receiveMultiPrimitiveMessages()");
         abort_nolock();
     }
 
 out:
 
-    if (++recvExited == fNumThreads)
+    // Take just the first one.
+    auto data = getJoinLocalDataByIndex(0);
+    // We have only one thread here.
     {
-
         if (doJoin && smallOuterJoiner != -1 && !cancelled())
         {
+            vector<rowgroup::RGData> rgDatav;
             tplLock.unlock();
```

Comment:
If the lock is really needed(plz take a look at the prev comment) we better move vector ctor outside the crit section.